src/main/java/com/dabsquared/gitlabjenkins/webhook/build/MergeRequestBuildAction.java

```diff
@@ -29,7 +29,12 @@ public MergeRequestBuildAction(Job<?, ?> project, String json) {
         this.mergeRequestHook = JsonUtil.read(json, MergeRequestHook.class);
     }
 
-    public void execute(StaplerResponse response) {
+    /**
+    * Noop for merge request hooks.
+    */
+    void processForCompatibility() {}
```

Comment:
As this method isn't implemented for a merge request why does the MergeRequestBuildAction extend the BuildWebHookAction at all? This class still just needs to implement the WebHookAction interface. This will make the  BuildWebHookAction obsolete as you can move the processForCompatibilty stuff directly to the PushBuildAction.